repository/src/main/java/org/apache/atlas/tasks/TaskRegistry.java

```diff
@@ -204,6 +204,32 @@ public void complete(AtlasVertex taskVertex, AtlasTask task) {
         updateStatus(taskVertex, task);
     }
 
+    @GraphTransaction
+    public void softDelete(String guid) throws AtlasBaseException{
+        try {
+            AtlasGraphQuery query = graph.query()
+                    .has(Constants.TASK_TYPE_PROPERTY_KEY, Constants.TASK_TYPE_NAME)
+                    .has(TASK_GUID, guid);
+
+            Iterator<AtlasVertex> results = query.vertices().iterator();
+
+            if (results.hasNext()) {
+                AtlasVertex taskVertex = results.next();
+
+                setEncodedProperty(taskVertex, Constants.TASK_STATUS, AtlasTask.Status.DELETED);
+                setEncodedProperty(taskVertex, Constants.TASK_UPDATED_TIME, System.currentTimeMillis());
+            }
+        }
+        catch (Exception exception) {
+            LOG.error("Error: on soft delete: {}", guid);
+
+            throw new AtlasBaseException(exception);
+        }
+        finally {
+            graph.commit();
```

Comment:
I don't think we need manually commit the changes.
Can you please verify this once
Also check GraphTransactionInterceptor class

Reply:
You are right, doCommit automatically invoked